modules/extension/app-schema/app-schema/src/main/java/org/geotools/appschema/jdbc/JoiningJDBCFeatureSource.java

```diff
@@ -470,10 +465,7 @@ public String selectSQL(
                 }
 
                 joinClause.append(" = ");
-                String fromTypeName;
-                if (isCount && join.getJoinedTypeName() != null)
-                    fromTypeName = join.getJoinedTypeName();
-                else fromTypeName = curTypeName;
+                String fromTypeName = curTypeName;
```

Comment:
Question: why we remove the joined type name use case?

Reply:
introduced by me in a previous commit related to the count functionalities. It was not actually needed so I'm removed